src/sklearn_compat/_sklearn_compat.py

```diff
@@ -206,8 +206,56 @@ def _is_fitted(estimator, attributes=None, all_or_any=all):
         ]
         return len(fitted_attrs) > 0
 
+    if sklearn_version < parse_version("1.3"):
+        # only define those functions to not raise an error.
+        def process_routing(_obj, _method, /, **kwargs):
+            pass
+
+        def _raise_for_params(params, owner, method):
+            pass
```

Comment:
I'm okay for `_raise_for_params` to do nothing, but `process_routing` shouldn't exist. It will 100% result in running invalid code and raising errors later in the code.

Reply:
So let me raise a proper error here mentioning that this is only available in 1.3 at least.